filesystem/src/main/java/com/nytimes/android/external/fs/filesystem/FSFile.java

```diff
@@ -71,6 +71,9 @@ public void write(BufferedSource source) throws IOException {
             if (sink != null) {
                 sink.close();
             }
+            if (source != null) {
+                source.close();
+            }
```

Comment:
Why the null check? Makes more sense to me to annotate the source param as `@NonNull`. The whole method makes no sense with a null source and it should throw an exception if it's called with wrong arguments.